libs/src/services/identity/index.js

```diff
@@ -105,14 +105,21 @@ class IdentityService {
    * @param {number} trackId
    * @param {number} userId
    */
-  async logTrackListen (trackId, userId) {
-    return this._makeRequest({
+  async logTrackListen (trackId, userId, listenerAddress) {
+    const request = {
       url: `/tracks/${trackId}/listen`,
       method: 'post',
       data: {
         userId: userId
       }
-    })
+    }
+
+    if (listenerAddress) {
+      request.headers = {
+        'x-forwarded-for': listenerAddress
```

Comment:
when identity parses this header, won't it take the "left most" x-forwarded-for, which will be the creator node's IP rather than this forwarded one? (I don't think testing this locally is enough b/c of the lack of proxies, if you cut the branch with the term "hotfix" anywhere in it, CI will build an image and you can release that to a staging cnode and verify)